newrelic/hooks/external_botocore.py

```diff
@@ -889,7 +891,24 @@ def _nr_sqs_message_trace_wrapper_(wrapped, instance, args, kwargs):
     return _nr_sqs_message_trace_wrapper_
 
 
+def wrap_lambda_invoke(wrapped):
+    def _nr_wrap_lambda_invoke(wrapped, instance, args, kwargs):
+        transaction = current_transaction()
+        if not transaction:
+            return wrapped(*args, **kwargs)
+
+        bound_args = bind_args(wrapped, args, kwargs)
+        arn = bound_args["kwargs"].get("FunctionName")
+        if arn and hasattr(instance, "meta") and hasattr(instance.meta, "events"):
```

Comment:
This is just an object that I found that is also available inside the make_request. It's possible there is something better to attach this to but it was hard to find something that was accessible in both spots. I didn't want to attach it to the transaction as I thought there might be a conflict with that in async cases.